core/templates/pages/topics-and-skills-dashboard-page/skills-list/assign-skill-to-topic-modal.controller.spec.ts

```diff
@@ -0,0 +1,46 @@
+// Copyright 2020 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Unit tests for AssignSkillToTopicModalController.
+ */
+
+describe('Assign Skill To Topic Modal ', function() {
+  var $scope = null;
+  var $uibModalInstance = null;
+  var topicDict = {
+    id: '1',
+    name: 'name',
+    description: 'topic description',
+  };
+  beforeEach(angular.mock.module('oppia'));
+  beforeEach(angular.mock.inject(function($injector, $controller) {
+    var $rootScope = $injector.get('$rootScope');
+
+    $uibModalInstance = jasmine.createSpyObj(
+      '$uibModalInstance', ['close', 'dismiss']);
+
+    $scope = $rootScope.$new();
+    $controller('AssignSkillToTopicModalController', {
+      $scope: $scope,
+      $uibModalInstance: $uibModalInstance,
+      topicSummaries: [topicDict]
```

Comment:
The naming seems wrong. A topic is not the same as a topic summary.

Also, shouldn't you be working with frontend domain objects here, not dicts?

Reply:
Done.